### ISSUE TITLE
How to use the UI

### ISSUE THREAD

**Issue by 1tianmaxingkong1** [nvdungx/HLC_TestTool#1]
I am very interested in this project and thank the author for his contribution, but I don't know much about the technology applied in this project, can you explain how the project works?

**Comment by nvdungx** [comment]
Hi,
A brief description of this repos:
The target of this project is to build a conformance test environment for ISO15118-2(V2G section) specification following the sample TTCN-3 test content, which is available in ISO15118-4.
About the ISO15118 spec, which is essentially a protocol spec that defines communication between EV and EV Charging station, you can search and read it online.

1. For the hardware:
- a Raspberry Pi4B and an extension board(with Qualcomm qca7000 for physical interface of power line communication, some relay+resistor and stm32f0 for basic signalling control)
- interfacing with qca7000(kind of like ethernet PHY transceiver) required some setting changes in the Raspberry, you can search for it on the internet and refer to this repos https://github.com/qca/open-plc-utils

2. For the software: << heavily unoptimized and  hosted in the Raspberry 
- build system: make, gcc/g++, and Cmake. search around and you will know how to build it.
- used libraries(open-source): << install all of them then it should able to build, some already available in "modules" folder
  - openv2g: for v2g msg encoding/decoding
  - boost-asio: for runtime operation
  - cppcommon: tcp/udp from https://github.com/chronoxor
  - rapidjson: parsing some data
  - fmt: string format
  - pigpio: control pi gpio
- source code: all other directories except the "modules", the implementation is based on TTCN-3 code in ISO spec.

Basically, the system shall operate as a test device (EV emulation) to verify the operation(mostly communication) of SUT(which is an EV charging station)

The other repos TestUI is just a simple web-base tool, which shall be hosted on the Raspberry too and it shall work in conjunction with this repos to create a testing environment. User can access, execute test by accessing the webUI

**Comment by 1tianmaxingkong1** [comment]
Hi NvDung:
        
Glad to receive your reply!
We are now developing EVCC controller. We want to test the controller and find a simulated charging pile system. You mentioned in your reply that the system is to verify the consistency communication test of the charging pile. Do you plan to implement the consistency test system of the charging controller at the vehicle end, or can this system be easily changed?
Look forward to your reply!

good luck!
    



祝好！
常俊杰 

 
天津优控智行科技有限公司 嵌入式软件工程师
***@***.***
网址：www.eco-ev.com 
手机：15900398565
地址：天津市华苑产业园区开华道22号普天创新园东塔1702
 
From: NvDung
Date: 2022-11-05 01:53
To: nvdungx/ISO15118_4_TestEnv
CC: 1tianmaxingkong1; Author
Subject: Re: [nvdungx/ISO15118_4_TestEnv] How to use the UI (Issue #1)
Hi,
A brief description of this repos:
The target of this project is to build a conformance test environment for ISO15118-2(V2G section) specification following the sample TTCN-3 test content, which is available in ISO15118-4.
About the ISO15118 spec, which is essentially a protocol spec that defines communication between EV and EV Charging station, you can search and read it online.
For the hardware:
a Raspberry Pi4B and an extension board(with Qualcomm qca7000 for physical interface of power line communication, some relay+resistor and stm32f0 for basic signalling control)
interfacing with qca7000(kind of like ethernet PHY transceiver) required some setting changes in the Raspberry, you can search for it on the internet and refer to this repos https://github.com/qca/open-plc-utils
For the software: << heavily unoptimized and hosted in the Raspberry
build system: make, gcc/g++, and Cmake. search around and you will know how to build it.
used libraries(open-source): << install all of them then it should able to build 
openv2g: for v2g msg encoding/decoding
boost-asio: for runtime operation
cppcommon: tcp/udp from https://github.com/chronoxor
rapidjson: parsing some data
fmt: string format
pigpio: control pi gpio
source code: all other directories except the "modules", the implementation is based on TTCN-3 code in ISO spec.
Basically, the system shall operate as a test device (EV emulation) to verify the operation(mostly communication) of SUT(which is an EV charging station)
The other repos TestUI is just a simple web-base tool, which shall be hosted on the Raspberry too and it shall work in conjunction with this repos to create a testing environment. User can access, execute test by accessing the webUI

**Comment by 1tianmaxingkong1** [comment]
Hi NvDung:
        
Glad to receive your reply!
We are now developing EVCC controller. We want to test the controller and find a simulated charging pile system. You mentioned in your reply that the system is to verify the consistency communication test of the charging pile. Do you plan to implement the consistency test system of the charging controller at the vehicle end, or can this system be easily changed?
Look forward to your reply!

good luck!

**Comment by nvdungx** [comment]
Hi,

Currently, this environment only supports verifying the SECC side. For the vehicle side, it requires additional implementation of the logic to verify messages from EVCC.
Yes, I intend to implement the rest of ISO15118-4, and ISO15118-5, which include test suites for the vehicle side, plug & charge functions, and datalink layer(PLC) but it's pending for now since I don't have much time recently.

**Comment by 1tianmaxingkong1** [comment]
Hi,
Glad to hear from you!
We are currently very interested in the project, is this possible if I want to join you in the development work? If so, what competencies would I need to have and what preparation would I need to do?
Looking forward to your dafu

**Comment by nvdungx** [comment]
Hi,
Yes, feel free to fork or contribute directly to this project. On competencies, C++, linux, basic knowledge of ethernet is required, ISO15118 specification series as well.
If you want to pick up this project from its current status, please contact me through email or msteam at nvdung694@gmail.com, I will guide you through it
